### DISCUSSION TITLE
Docker pull from public GitHub Package Registry fail with "no basic auth credentials" error

### DISCUSSION THREAD

**Discussion by xmlking** [community/community#25689] [Packages]
<p>I pushed docker image to my public repo.</p>
<p>from other computer, I tried to pull, and got -<em>no basic auth credentials- error</em></p>
<p>Do I have to login even for pulling public images ?</p>
<p>docker pull <a href="http://docker.pkg.github.com/xmlking/micro-starter-kit/account-srv:v0.1.1" class="inline-onebox-loading" rel="nofollow noopener">docker.pkg.github.com/xmlking/micro-starter-kit/account-srv:v0.1.1</a><br>
Error response from daemon: Get <a href="https://docker.pkg.github.com/v2/xmlking/micro-starter-kit/account-srv/manifests/v0.1.1" rel="nofollow noopener">https://docker.pkg.github.com/v2/xmlking/micro-starter-kit/account-srv/manifests/v0.1.1</a>: no basic auth credentials</p>

**Comment by ethomson** [comment, marked as answer]
<p>Yes, at present you do need to authenticate to read even publicly available packages.</p>

**Comment by bensheldon** [comment]
<p>How would I go about referencing a GitHub registry hosted docker package within an <code>action.yml</code> for a GitHub Action so that it contained auth credentials? e.g.</p>
<pre><code># ...
runs:
  using: docker
  image: "docker://docker.pkg.github.com/bensheldon/action-add-to-project/action-add-to-project:latest"
</code></pre>

**Comment by Sauci** [comment]
<p>Hi, got the same issue, even if I authenticate, here is my yml file content:</p>
<pre><code>...  
...  
jobs:
publishDockerImageToRegistry:
name: Publish Docker image to registry
runs-on: ubuntu-latest
steps:
-uses: actions/checkout@master
-name: Publish to registry
uses: elgohr/Publish-Docker-Github-Action@master
with:
name: docker.pkg.github.com/ ***/*** /***
username: ${{secrets.GITHUB_PKG_REGISTRY_USERNAME}}
password: ${{secrets.GITHUB_PKG_REGISTRY_PASSWORD}}
registry: docker.pkg.github.com
buildFirmware:
needs: publishDockerImageToRegistry
name: Build bootloader firmware
runs-on: ubuntu-latest
strategy:
matrix:
toolchain: [gnu, llvm]
steps:
-uses: actions/checkout@v1
with:
submodules: true
-uses: actions/docker/login@master
env:
DOCKER\_REGISTRY\_URL: docker.pkg.github.com
DOCKER\_USERNAME: ${{secrets.GITHUB_PKG_REGISTRY_USERNAME}}
DOCKER\_PASSWORD: ${{secrets.GITHUB_PKG_REGISTRY_PASSWORD}}
-name: Pull Docker image
run: docker pull docker.pkg.github.com/ ***/*** /***:latest
-name: Configure build system
...
...
</code></pre>
<p> I can successfully publish image to registry, but can’t pull it in step ‘Pull Docker image’. Am I missing something?</p>
<p>Thanks</p>

**Comment by farshidtz** [comment]
<p>Hi, Is this a bug? For public images, the layers can be downloaded from the GUI without login.</p>

**Comment by p3lim** [comment]
<p>This makes it very hard to use the public registry, it’ll need to not require auth if you want any real adoptation.</p>

**Comment by lucacasonato** [comment]
<p>Can this be changed please? There is no good reason to make this be the case as you can download from the site even without login. This blocks all kinds of third party CI services from working as often you can not specify registry credentials for those.</p>

**Comment by Starefossen** [comment]
<p>Can this limitation be written with bold letters in the documentation becuase it makes GitHub Package Registry not usable for any open source projects at the moment.</p>

**Comment by danacr** [comment]
<p>Are there any plans on removing this requirement? Currently it is not possible for us to use the docker registry in any of our projects. This was one of the main reasons we thought we could use GitHub.</p>

**Comment by jfri3d** [comment]
<p>This. Completely this. The lack of public non-auth pulls is killing any sort of “draw” to GitHub Packages…</p>

**Comment by abatilo** [comment]
<p>Is it planned that this will change?</p>

**Comment by ethomson** [comment]
<p>Thanks for the feedback.  We realize that this is a popular feature request, and we’ve passed it along to the GitHub Package Registry team.</p>

**Comment by RothAndrew** [comment]
<p>Any update on this? I’d love to have everything live on GitHub, but without being able to pull the image without logging in, I’m going to have to publish to Docker Hub instead.</p>

**Comment by kellervater** [comment]
<p>You’re right. And for this purpose I did a workaround you can use to publish your freshly built docker images to Dockerhub.<br>
<a href="https://github.com/saubermacherag/gpr-docker-publish" rel="nofollow noopener">https://github.com/saubermacherag/gpr-docker-publish</a><br>
this is a fork of a fork of a github action (without release currently).<br>
It builds and publishes docker images to GPR except you use the param <code>DOCKERHUB_REPOSITORY</code>.<br>
Maybe that helps until Github comes up with a solution.</p>

**Comment by emilingerslev** [comment]
<p><a class="mention" href="/ethomson">@ethomson</a> do you have any way to follow along with the issue? A ticket or something?</p>

**Comment by iBobik** [comment]
<p>GitLab Docker registry does not require authentication for public projects.</p>
<p><a href="https://docs.gitlab.com/ee/user/packages/container_registry/index.html#using-with-private-projects" rel="nofollow noopener">https://docs.gitlab.com/ee/user/packages/container_registry/index.html#using-with-private-projects</a></p>

**Comment by nightlark** [comment]
<p>This is a problem I ran into when trying to make an action that runs a script in a container that I have published to GitHub Package Registry. Unfortunately it fails on the <code>FROM</code> statement because authentication is required, making Package Registry unuseable as a place to provide the base layer for a Docker action.</p>

**Comment by fforootd** [comment]
<p>Any news on this. It is kind of a big issue. </p>

**Comment by stefan-kiss** [comment]
<p>i just tried this feature.<br>
I have to say i am disapointed first for the lack of transparency. Yes there are tutorials on how to login, but then again all public repositories support unauthenticated downloads.</p>
<p>Why is it called public docker registry if you need authentication AND permissions ?<br>
So please first fix the documentation. You need to specify this very clear from the begining. Maybe even change the feature’s name. Private docker registry. Private packages.</p>
<p>Second … well it would be nice if we could publicly read from a public docker registry ?</p>

**Comment by Burnett01** [comment]
<ul>
<li>Push -</li>
</ul>
<p>Any news on this, please?</p>
